sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatViewModel.kt

```diff
@@ -1194,6 +1194,26 @@ abstract class ChatViewModel<ARGS: NavArgs>(
             )
         }
     }
+
+    private var payAttachmentJob: Job? = null
+    suspend fun payAttachment(message: Message) {
+        if (payAttachmentJob?.isActive == true) {
+            return
+        }
+
+        payAttachmentJob = viewModelScope.launch(mainImmediate) {
```

Comment:
the job block does nothing here as it is reliant on the subscription collection (not the `AlertConfirmPayAction` lambda). As the dialogue builder will be inhibiting any other user interaction, would it be better to set `payAttachmentJob` to the `AlertConfirmPayAction`'s coroutine, or just remove all together?

Reply:
I moved the payAttachmentJob inside the AlertConfirmPayAction callback, but I kept it since the UI will update once the payAttachment request returns and the purchase item is upserted (showing the processing state). It's just a second or less, but during that time user shouldn't be able to purchase item again.